handler.go

```diff
@@ -431,28 +434,37 @@ func (fm *Frontman) processInputContinuous(inputFilePath string, local bool, int
 	lastFetch := time.Unix(0, 0)
 	interval := secToDuration(fm.Config.Sleep)
 
+	waitWg := sync.WaitGroup{}
+
+	logrus.Infof("STARTING processInputContinuous")
+
 	for {
+		select {
+		case <-interrupt:
+			logrus.Infof("processInputContinuous got interrupt,waiting & stopping")
+			waitWg.Wait()
+			//close(*resultsChan)
+			return
+		default:
+		}
+
 		if time.Since(lastFetch) >= interval {
+			logrus.Infof("processInputContinuous running updateInputChecks")
 			lastFetch = time.Now()
 			go fm.updateInputChecks(inputFilePath)
 		}
 
-		if len(fm.checks) > 0 {
-			if currentCheck, ok := fm.takeNextCheckNotInProgress(); ok {
-				fm.ipc.add(currentCheck.uniqueID())
+		if currentCheck, ok := fm.takeNextCheckNotInProgress(); ok {
+			fm.ipc.add(currentCheck.uniqueID())
 
-				go func(check Check, results *chan Result, inProgress *inProgressChecks) {
-					res, _ := fm.runCheck(check, local)
-					inProgress.remove(check.uniqueID())
-					*results <- *res
-				}(currentCheck, resultsChan, &fm.ipc)
-			}
-		}
-		select {
-		case <-interrupt:
-			close(*resultsChan)
-			return
-		default:
+			go func(check Check, results *chan Result, inProgress *inProgressChecks, wg *sync.WaitGroup) {
+				wg.Add(1)
+				defer wg.Done()
+
+				res, _ := fm.runCheck(check, local)
```

Comment:
is it ignoring the error?

Reply:
No. Errors gets encoded in the result here

https://github.com/cloudradar-monitoring/frontman/blob/DEV-1757b/handler.go#L531

Reply:
it's not a clean way I would say but ok